amm/contracts/stable_pool/lib.rs

```diff
@@ -718,7 +723,7 @@ pub mod stable_pool {
             let current_time = self.env().block_timestamp();
             let mut rate_changed = false;
             for rate in self.pool.token_rates.iter_mut() {
-                rate_changed = rate_changed || rate.update_rate_no_cache(current_time);
+                rate_changed = rate.update_rate_no_cache(current_time) || rate_changed;
```

Comment:
I'd prefer `|` there anyway, as (1) Using `||` in expressions used in assignments is usually asking for troubles (someone might fork the repo and add some new condition after `rate_changed`, which would be bad), and  (2) I'm not sure if the gas cost of `||` wouldn't be greater, because of conditional jumps (probably in nanoscale, but still).
But it's correct and contracts are non-upgradeable, so up to you :P

Reply:
I personally like `||` more than bitwise `|`
* It's more clear
* When using `|`, `rate_changed` would have to have a number type (`|` shouldn't be applicable to `bool`) and this is weird
* Optimization of `|` sounds negligible

I can also propose an alternative which is 100% clear :stuck_out_tongue: 
```rust
if rate.update_rate_no_cache(current_time) {
  rate_changed = true;
}
```